crypto/fipsmodule/bn/montgomery.c

```diff
@@ -477,15 +477,29 @@ static void montgomery_s2n_bignum_mul_mont(BN_ULONG *rp, const BN_ULONG *ap,
   uint64_t w = n0[0];
 
   if (num == 32) {
+#if defined(__ARM_NEON)
```

Comment:
It's preferable to replace the occurrences of `#if defined` with `if CRYPTO_is_NEON_capable()`. That's possible since this code is surrounded by `BN_MONTGOMERY_S2N_BIGNUM_CAPABLE`, which is defined only for AArch64.
We could have assumed that neon is always available on AArch64, but in FIPS tests, we disable all capabilities as one of the configurations we test. So we need this check and this choice between vectorised and non-vectorised implementations.